src/Functions/FunctionsJSON.cpp

```diff
@@ -803,25 +821,26 @@ struct JSONExtractTree
         }
     };
 
-    class LowCardinalityNode : public Node
+    class LowCardinalityFixedStringNode : public Node
     {
     public:
-        LowCardinalityNode(DataTypePtr dictionary_type_, std::unique_ptr<Node> impl_)
-            : dictionary_type(dictionary_type_), impl(std::move(impl_)) {}
+        LowCardinalityFixedStringNode(const size_t fixed_length_)
+            : fixed_length(fixed_length_) {}
         bool insertResultToColumn(IColumn & dest, const Element & element) override
         {
-            auto from_col = dictionary_type->createColumn();
-            if (impl->insertResultToColumn(*from_col, element))
-            {
-                std::string_view value = from_col->getDataAt(0).toView();
-                assert_cast<ColumnLowCardinality &>(dest).insertData(value.data(), value.size());
-                return true;
-            }
-            return false;
+            auto str = element.getString();
+
+            if (str.size() > fixed_length)
+                return false;
+
+            auto padded_str = str.data() + std::string(fixed_length - std::min(fixed_length, str.length()), '\0');
```

Comment:
Where did it come from? There was no padding prior to `FixedStringColumn::insertData()` calls. If it is some kind of optimization, please explain it in comments.

Reply:
Added, let me know if it's clear